src/adaBase58_test.c

```diff
@@ -0,0 +1,57 @@
+#include <os.h>
+#include <stdbool.h>
+#include "assert.h"
+#include "test_utils.h"
+#include "hex_utils.h"
+#include "adaBase58.h"
+#include "stream.h"
+
+void run_adaBase58_test()
+{
+#define TESTCASE(input_, expected_) \
+	{ \
+	    stream_t si; \
+	    stream_init(&si); \
+	    stream_appendFromHexString(&si, input_); \
+		uint8_t* inputBuffer = stream_head(&si); \
+		uint8_t inputBufferSize = stream_availableBytes(&si); \
+		stream_t se; \
+	    stream_init(&se); \
+	    stream_appendFromHexString(&se, expected_); \
+		uint8_t* expectedBuffer = stream_head(&se); \
+		uint8_t expectedBufferSize = stream_availableBytes(&se); \
+		uint8_t output[248]; \
+		uint8_t outputLength = ada_encode_base58(inputBuffer, inputBufferSize, output, sizeof(output)); \
+		EXPECT_EQ(outputLength, expectedBufferSize); \
```

Comment:
Note: I have `EXPECT_EQ_BYTES(ptr1, ptr2, len)` somewhere in the commit pipeline ...